lib/http.js

```diff
@@ -173,10 +173,12 @@ bedrock.events.on('bedrock-express.configure.routes', app => {
       const {keystore: existingConfig} = req.webkms;
       if(existingConfig.id !== req.body.id) {
         throw new BedrockError(
-          'Configuration "id" does not match.',
-          'DataError', {
+          'Configuration "id" does not match request URL.',
+          'URLMismatchError', {
```

Comment:
This definitely seems like a breaking change to me `DataError` -> `URLMismatchError`.  If we don't have a local/downstream test that asserts on this error type (at minimum) then it's  a gap in code coverage.

Reply:
This type of error could already be emitted (another code path was just combined) -- this is related to having more consistently retrieved configs.

Reply:
Is it possible that a consumer of this module has a test in a test suite, or other code that checks for a `DataError`, and by virtue of this change that test or code will break?

Does changing the type of an error constitute a breaking change or not?

Why would we *not* take the position that it is in fact a breaking change?

Reply:
I think the situation is this:

1. Some software would need to be misusing the update API to get an error here.
2. If that software was code branching off of the error, they already needed to handle errors of both type X and Y.

Therefore, I interpret this as removing one of those two possible errors so that one of those code branches never executes. Note that this is also not a "feature removal" as this was an error condition, not a feature.

Reply:
It also appears that the `DataError` was never even reachable. The conditions required to trigger that error would have been the same as those required to trigger the `URLMismatchError` -- and that error was first in the code path, so no one should have ever seen the `DataError`. It was dead code.